torch/nn/utils/_expanded_weights/conv_expanded_weights.py

```diff
@@ -0,0 +1,67 @@
+import torch
+import torch.nn.functional as F
+
+from .conv_utils import conv_backward, conv_args_and_kwargs
+from .expanded_weights_impl import implements_per_sample_grads
+from .expanded_weights_utils import forward_helper
+
+THRESHOLD = 257
+
+@implements_per_sample_grads(F.conv1d)
+class Conv1dPerSampleGrad(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, kwarg_names, *expanded_args_and_kwargs):
+        if any([isinstance(i, str) for i in expanded_args_and_kwargs]):
+            raise RuntimeError("Expanded Weights does not support convolution padding as a string. "
+                               "Please file an issue to prioritize support")
+        expanded_args, expanded_kwargs = conv_args_and_kwargs(kwarg_names, expanded_args_and_kwargs)
+        output = forward_helper(F.conv1d, expanded_args, expanded_kwargs)
+        ctx.input, ctx.weight = expanded_args
+        ctx.bias = expanded_kwargs['bias']
+        ctx.stride, ctx.padding = expanded_kwargs['stride'], expanded_kwargs['padding']
+        ctx.dilation, ctx.groups = expanded_kwargs['dilation'], expanded_kwargs['groups']
+        return output
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        return conv_backward(F.conv1d, ctx, grad_output)
+
```

Comment:
Can we de-duplicate this more?

```
class ConvPerSampleGrad(torch.autograd.Function):
    @staticmethod
    def forward(ctx, kwarg_names, *expanded_args_and_kwargs):
        if any([isinstance(i, str) for i in expanded_args_and_kwargs]):
            raise RuntimeError("Expanded Weights does not support convolution padding as a string. "
                               "Please file an issue to prioritize support")
        expanded_args, expanded_kwargs = conv_args_and_kwargs(kwarg_names, expanded_args_and_kwargs)
        conv_fn = get_conv_fn(expanded_args) # selects 1d, 2d, or 3d.
        output = forward_helper(conv_fn, expanded_args, expanded_kwargs)
        ctx.input, ctx.weight = expanded_args
        ctx.bias = expanded_kwargs['bias']
        ctx.stride, ctx.padding = expanded_kwargs['stride'], expanded_kwargs['padding']
        ctx.dilation, ctx.groups = expanded_kwargs['dilation'], expanded_kwargs['groups']
        return output

    @staticmethod
    def backward(ctx, grad_output):
        return conv_backward(conv_fn, ctx, grad_output)

# IDK if the decorator actually works like this
implements_per_sample_grads(F.conv1d)(ConvPerSampleGrad)
implements_per_sample_grads(F.conv2d)(ConvPerSampleGrad)
implements_per_sample_grads(F.conv3d)(ConvPerSampleGrad)
```

Reply:
Fixed--thankfully decorators what we would hope in that situation